server/server.js

```diff
@@ -220,11 +234,22 @@ Server.prototype._handleResourceMessage = function(socket, message, messageType)
     this._storeResource(resource);
     this._persistenceSubscribe(resource.name, socket.id);
     this._updateLimits(socket, message, resource.options);
+    this._stampMessage(socket, message);
```

Comment:
this stuff is good. But keep off this PR please. Cherry pick this commit out of this branch into new one and PR that if you want. But let's keep focus on the actual change. I am finding it hard to review all these giant PRs.